packages/cosmic-swingset/test/run-policy.test.js

```diff
@@ -0,0 +1,200 @@
+  DEFAULT_SIM_SWINGSET_PARAMS,
+  makeVatCleanupBudgetFromKeywords,
+} from '../src/sim-params.js';
+
+/** @import { KVStore } from '../src/helpers/bufferedStorage.js' */
+
+/**
+ * Converts a Record<name, specifierString> into Record<name, { sourceSpec }>
+ * for defining e.g. a `bundles` group.
+ *
+ * @param {Record<string, string>} src
+ * @returns {import('@agoric/swingset-vat').SwingSetConfigDescriptor}
+ */
+const makeSourceDescriptors = src => {
+  const hardened = objectMap(src, sourceSpec => ({ sourceSpec }));
+  return JSON.parse(JSON.stringify(hardened));
+};
+
+/**
+ * @param {import('../src/sim-params.js').VatCleanupKeywordsRecord} budget
+ * @returns {import('@agoric/cosmic-proto/swingset/swingset.js').ParamsSDKType}
+ */
+const makeCleanupBudgetParams = budget => {
+  return {
+    ...DEFAULT_SIM_SWINGSET_PARAMS,
+    vat_cleanup_budget: makeVatCleanupBudgetFromKeywords(budget),
+  };
+};
+
+test('cleanup work must be limited by vat_cleanup_budget', async t => {
+  const { toStorage: handleVstorage } = makeFakeStorageKit('');
+  const receiveBridgeSend = (destPort, msg) => {
+    switch (destPort) {
+      case BridgeId.STORAGE: {
+        return handleVstorage(msg);
+      }
+      default:
+        Fail`port ${q(destPort)} not implemented for message ${msg}`;
+    }
+  };
+  const { pushCoreEval, runNextBlock, shutdown, swingStore } =
+    await makeCosmicSwingsetTestKit(receiveBridgeSend, {
+      bundles: makeSourceDescriptors({
+        puppet: '@agoric/swingset-vat/tools/vat-puppet.js',
+      }),
+      configOverrides: {
+        // Ensure multiple spans and snapshots.
+        defaultManagerType: 'xsnap',
+        snapshotInitial: 2,
+        snapshotInterval: 4,
+      },
+      fixupBootMsg: () => ({
+        ...defaultBootstrapMessage,
+        params: makeCleanupBudgetParams({ Default: 0 }),
+      }),
+    });
+  const mapStore = provideEnhancedKVStore(
+    /** @type {KVStore<string>} */ (swingStore.kernelStorage.kvStore),
+  );
+  /** @type {(key: string) => string} */
+  const mustGet = key => {
+    const value = mapStore.get(key);
+    assert(value !== undefined, `kvStore entry for ${key} must exist`);
+    return value;
+  };
+
+  // Launch the new vat and capture its ID and store snapshotting.
+  pushCoreEval(async powers => {
+    const { bootstrap } = powers.vats;
+    await E(bootstrap).createVat('doomed', 'puppet');
+  });
+  await runNextBlock();
+  const vatIDs = JSON.parse(mustGet('vat.dynamicIDs'));
+  const vatID = vatIDs.at(-1);
+  t.is(
+    vatID,
+    'v8',
+    `time to update expected vatID to ${JSON.stringify(vatIDs)}.at(-1)?`,
+  );
+  const getKV = () =>
+    [...mapStore].filter(([key]) => key.startsWith(`${vatID}.`));
+  // console.log(swingStore.kernelStorage);
+
+  t.false(
+    JSON.parse(mustGet('vats.terminated')).includes(vatID),
+    'must not be terminated',
+  );
+  const initialEntries = new Map(getKV());
+  t.regex(
+    String(initialEntries.size),
+    /^[1-9][0-9]*$/,
```

Comment:
Out of curiosity, why not

```suggestion
  t.not(
    initialEntries.size,
    0,
```

Reply:
Yeah, I like that better. The important thing was ensuring that the actual value is visible in case of failure.